hedera-node/src/main/java/com/hedera/services/throttling/DeterministicThrottling.java

```diff
@@ -288,15 +425,183 @@ private boolean shouldThrottleMint(ThrottleReqsManager manager, TokenMintTransac
-				(gasThrottle == null || !gasThrottle.allow(now, accessor.getGasLimitForContractTx()));
+				isGasThrottled(function) &&
+				(gasThrottle == null || !gasThrottle.allow(now,
+						isChild ? MiscUtils.getGasLimitForContractTx(txn, function, null)
+								: accessor.getGasLimitForContractTx()));
+	}
+
+	private void reclaimLastAllowedUse() {
+		activeThrottles.forEach(DeterministicThrottle::reclaimLastAllowedUse);
+		if (gasThrottle != null) {
+			gasThrottle.reclaimLastAllowedUse();
+		}
+	}
+
+	private void resetLastAllowedUse() {
+		activeThrottles.forEach(DeterministicThrottle::resetLastAllowedUse);
+		if (gasThrottle != null) {
+			gasThrottle.resetLastAllowedUse();
+		}
+	}
+
+	private void calculateThrottles(ThrottleDefinitions defs, long capacitySplit) {
+		List<DeterministicThrottle> newActiveThrottles = new ArrayList<>();
+		EnumMap<HederaFunctionality, List<Pair<DeterministicThrottle, Integer>>> reqLists
+				= new EnumMap<>(HederaFunctionality.class);
+
+		for (var bucket : defs.getBuckets()) {
+			try {
+				var mapping = bucket.asThrottleMapping(capacitySplit);
+				var throttle = mapping.getLeft();
+				var reqs = mapping.getRight();
+				for (var req : reqs) {
+					reqLists.computeIfAbsent(req.getLeft(), ignore -> new ArrayList<>())
+							.add(Pair.of(throttle, req.getRight()));
+				}
+				newActiveThrottles.add(throttle);
+			} catch (IllegalStateException badBucket) {
+				log.error("When constructing bucket '{}' from state: {}", bucket.getName(), badBucket.getMessage());
+			}
+		}
+		EnumMap<HederaFunctionality, ThrottleReqsManager> newFunctionReqs = new EnumMap<>(HederaFunctionality.class);
+		reqLists.forEach((function, reqs) -> newFunctionReqs.put(function, new ThrottleReqsManager(reqs)));
+
+		functionReqs = newFunctionReqs;
+		activeThrottles = newActiveThrottles;
+		activeDefs = defs;
+	}
+
+	private void calculateScheduleThrottles(final ThrottleDefinitions defs) {
+
+		// copy the throttles
+		var defsCopy = ThrottleDefinitions.fromProto(defs.toProto());
+
+		// get the throttle groups with the minimum tps value for each transaction type,
+		// these are the effective max tps for each type of transaction
+		final EnumMap<HederaFunctionality, ThrottleGroup> minMtps = new EnumMap<>(HederaFunctionality.class);
+
+		for (var bucket : defsCopy.getBuckets()) {
+			for (var group : bucket.getThrottleGroups()) {
+				for (var op : group.getOperations()) {
+					// remove functions that can't be scheduled
+					if (MiscUtils.isSchedulable(op)) {
+						ThrottleGroup min = minMtps.get(op);
+						if (min == null || min.impliedMilliOpsPerSec() > group.impliedMilliOpsPerSec()) {
+							minMtps.put(op, group);
+						}
+					}
+				}
+			}
+		}
+
+		// dedup the min tps groups
+		final IdentityHashMap<ThrottleGroup, ThrottleBucket> groups = new IdentityHashMap<>();
+		for (var grp : minMtps.values()) {
+			groups.put(grp, null);
+		}
+
+		// filter out throttle groups we no longer need and set burst periods to 1
```

Comment:
I don't follow why we want to clamp all burst periods to 1; to limit scheduled transactions to some fraction `1/f` of the capacity for normal transactions, why not just take these modified buckets and do `asThrottleMapping(f)`?

(The burst periods will be auto-scaled to ensure every operation group can accept a transaction.)

Reply:
I was finding that the auto scaling overflows very easily and causes bounds checks elsewhere to throw.

I never really followed the internal math for the buckets so I didn't dare touch them. 

Setting the burst to 1 was the most straight forward method to say "this exact ops per second" that I got to work well without errors.